logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/ConversationEventReceiver.kt

```diff
@@ -242,20 +242,21 @@ internal class ConversationEventReceiverImpl(
         }
     }
 
-    private fun updateAssetMessage(message: Message.Regular, newMessageRemoteData: AssetContent.RemoteData): Message {
-        val assetMessageContent = message.content as MessageContent.Asset
+    private fun updateFinalAssetMessage(persistedMessage: Message.Regular, newMessageRemoteData: AssetContent.RemoteData): Message {
```

Comment:
Not sure that adding `Final` makes it more clear what this method does. 

Maybe something like this instead:
```suggestion
    private fun updateAssetMessageWithDecryptionKeys(persistedMessage: Message.Regular, newMessageRemoteData: AssetContent.RemoteData): Message {
```

Reply:
renamed